std/tai64.zig

```diff
@@ -0,0 +1,97 @@
+/// Support for TAI64 https://cr.yp.to/libtai/tai64.html
+
+const std = @import("index.zig");
+const builtin = @import("builtin");
+const mem = std.mem;
+const time = std.os.time;
+
+const TAI64N_EPOCH: u64 = 0x400000000000000a;
+
+/// TAI64N
+/// Use public methods to interact with internal 12 byte TAI64N format
+pub const TAI64N = struct {
+    // A TAI64 label is an integer between 0 and 2^64 referring to a
+    // particular second of real time. Integer s refers to the TAI second
+    // beginning exactly 2^62 - s seconds before the beginning of 1970 TAI,
+    // if s is between 0 inclusive and 2^62 exclusive; or the TAI second
+    // beginning exactly s - 2^62 seconds after the beginning of 1970 TAI, if
+    // s is between 2^62 inclusive and 2^63 exclusive. Integers 2^63 and
+    // larger are reserved for future extensions. Under many cosmological
+    // theories, the integers under 2^63 are adequate to cover the entire
+    // expected lifetime of the universe; in this case no extensions will be
+    // necessary. A TAI64 label is normally stored or communicated in
+    // external TAI64 format, consisting of eight 8-bit bytes in big-endian
+    // format. This means that bytes b0 b1 b2 b3 b4 b5 b6 b7 represent the
+    // label b0 * 2^56 + b1 * 2^48 + b2 * 2^40 + b3 * 2^32 + b4 * 2^24 + b5 *
+    // 2^16 + b6 * 2^8 + b7.
+
+    // A TAI64N label refers to a particular nanosecond of real time.
+    // It has two parts: 1) a TAI64 label and 2) an integer, between 0
+    // and 999999999 inclusive, counting nanoseconds from the
+    // beginning of the second represented by the TAI64 label.
+    tai64n: [12]u8,
```

Comment:
Why store this encoded? Seems like this should be:

```zig
sec: u64,
nsec: u30,
```

`to_bytes` can be:

```zig
pub fn encode(self: TAI64N) [12]u8 {
    var result: [12]u8 = undefined;
    mem.writeInt(result[0..8], self.sec, builtin.Endian.Big);
    mem.writeInt(result[8..12], self.nsec, builtin.Endian.Big);
    return result;
}
```

Then we can have a corresponding `decode` function.

Reply:
Forgot to add, with #287 this would be guaranteed to not memcpy.

Reply:
It is used heavily as a wire format, so it's easy to create the TAI64N by simply grabbing it off the wire -- likewise for placing it back on the wire. The less CPU cycles the better.

Reply:
Also, I don't know if it is a wise idea to use it platform internal. to reiterate, TAI64N is a wire encoding format (think databases and packet headers).

Reply:
In general if you want to grab something off the wire and save cpu cycles, then you'll keep it encoded and never decode it. So you'll have a []u8 and we don't need to put it in a struct and give it a name. That's optimal and simpler. 

The reason you would want to put it in a struct and give it a name is to do something with the decoded information. For example, from_unixtime(), now(), to_unixtime() all deal with the decoded information. 

Equality comparison and ordering comparison can be done directly on the encoded bytes as you demonstrated, and as we do with utf8-encoded strings.